handlers/handlers.go

```diff
@@ -194,10 +195,131 @@ func (s *ServiceSet) serviceHTML(db *mbtiles.DB) handlerFunc {
 	}
 }
 
+type tileCoord struct {
+	z    uint8
+	x, y uint64
+}
+
+// extractTileCoord extracts and returns tile coordinates and an optional
+// extension from a byte slice and returns them. In case the byte slice is too
+// short or the number parsing cannot be performed, the returned error is
+// non-nil.  pcs is expected to be in order (z, y, x) when the parameter swapXY
+// is false, and in order (z, x, y) when swapXY is true. The extension is
+// extracted from the third element of pcs. Any elements in pcs beyond the
+// third are ignored.
+func extractTileCoord(pcs []string, swapXY bool) (tc tileCoord, ext string, err error) {
+	if len(pcs) < 3 {
+		err = fmt.Errorf("not enough elements to extract tile coordinates")
+		return
+	}
+	var z64 uint64
+	if z64, err = strconv.ParseUint(pcs[0], 10, 8); err != nil {
+		err = fmt.Errorf("cannot parse zoom level: %v", err)
+		return
+	}
+	tc.z = uint8(z64)
+	const (
+		errMsgParse = "cannot parse %s coordinate axis: %v"
+		errMsgOOB   = "%s coordinate (%d) is out of bounds for zoom level %d"
+	)
+	if tc.x, err = strconv.ParseUint(pcs[1], 10, 64); err != nil {
+		err = fmt.Errorf(errMsgParse, "first", err)
+		return
+	}
+	if tc.x >= (1 << z64) {
+		err = fmt.Errorf(errMsgOOB, "first", tc.x, tc.z)
+		return
+	}
+	s := pcs[2]
+	if l := strings.LastIndex(s, "."); l >= 0 {
+		s, ext = s[:l], s[l:]
+	}
+	if tc.y, err = strconv.ParseUint(s, 10, 64); err != nil {
+		err = fmt.Errorf(errMsgParse, "second", err)
+		return
+	}
+	if tc.y >= (1 << z64) {
+		err = fmt.Errorf(errMsgOOB, "second", tc.y, tc.z)
+		return
+	}
+	if swapXY {
+		tc.x, tc.y = tc.y, tc.x
+	}
+	return
+}
+
+// respondNoTileFound writes the default response for a non-existing tile of type f to w
+func respondNoTileFound(w http.ResponseWriter, f mbtiles.TileFormat) (int, error) {
```

Comment:
How about `tileNotFoundHandler`?  Was inspired by name of `NotFoundHandler` in `http` package, though the usage pattern is a little different.

Reply:
Good idea, will do.